### ISSUE TITLE
WorkBot created editionless works

### ISSUE THREAD

**Issue by george08** [internetarchive/openlibrary#44]
This is broken.

E.g. These were all created on Aug 19, 2010. ALL editionless works should either be fixed (to show the source edition), or deleted.

http://openlibrary.org/works/OL15344034W/Handel_en_bestuur
http://openlibrary.org/works/OL15344038W/L%27_economia_delle_citt%C3%A0
http://openlibrary.org/works/OL15344035W/La_econom%C3%ADa_de_las_ciudades
http://openlibrary.org/works/OL15344045W/Vita_e_morte_delle_grandi_grandi_citt%C3%A0
http://openlibrary.org/works/OL15344041W/Muerte_y_vida_de_las_grandes_ciudades
http://openlibrary.org/works/OL15344039W/Les_villes_et_la_richesse_des_nations
http://openlibrary.org/works/OL15344043W/Stadt_im_Untergang

**Comment by LeadSongDog** [comment]
Before creating new works, check for editions similar to the workless one. Fuzzy matches on author and title ought to be enough for an "is it one of these?" suggestion. The last thing needed is more duplicate work records.

**Comment by jessamynwest** [comment]
see GH-153 also

**Comment by LeadSongDog** [comment]
The examples above from @george08 seem to be translations that are later attached to works of a different-language title, as at https://openlibrary.org/books/OL15423224M/Les_villes_et_la_richesse_des_nations?b=2&a=1&_compare=Compare&m=diff 
This then left the orphaned work https://openlibrary.org/works/OL15344039W/Les_villes_et_la_richesse_des_nations with no editions. 
The orphan should probably be made into a redirect to https://openlibrary.org/works/OL93638W/Cities_and_the_wealth_of_nations

**Comment by LeadSongDog** [comment]
Workbot has not run since January 2012.

**Comment by hornc** [comment]
I am finding other editionless works seemingly created by their editions having been correctly moved elsewhere. Is it safe to delete these editionless works? I'm trying to script a bot to do this and want to know what checks (if any) it needs to make.

**Comment by LeadSongDog** [comment]
I doubt that full automatic is desirable, errors are too likely. Should have human oversight to decide which of several similar works to associate.

**Comment by LeadSongDog** [comment]
@hornc Rethinking, better perhaps to redirect but remove from regular search results if that can be done. We should try to treat work URLs as permalinks when feasible.

**Comment by mekarpeles** [comment]
@hornc can this be closed?

**Comment by hornc** [comment]
@mekarpeles This is a data clean up task, I have actually run a long running task to identify affected works from WorkBot's history, and there are still some more to process. I had put this on hold as there was more pressing work on orphans to finish, but I'd like to revisit this to not lose the work in progress.

TL;DR: Let's keep it open and assigned to me as a data clean up task.

**Comment by hornc** [comment]
Two more examples from #44:
http://openlibrary.org/works/OL3536507W/Hong_lou_meng?v=20
https://openlibrary.org/works/OL16252229W/Kinder_des_weltalls?v=4

**Comment by mekarpeles** [comment]
I am less concerned with those 10 records -- I've noted them in a targeted issue #2727. I'm more concerned about what we're trying to accomplish with the issue.

It sounds like the issue is, Open Library allows Works with 0 editions.
And the question @george08 seems to be asking is:

What should happen (automatically) to a work when it has 0 editions; should it be deleted / cleaned up? Is this what normally happens?

**Comment by hornc** [comment]
Actually, this is a specific clean up task that affects very many edits done at scale by work bot in the past when merging translations. I have a script that has gone through and identified from the bot edit logs what was left hanging, and specifically what works they should be correctly redirected to. 

It has never been prioritised as an important task, and is why I haven't finished it -- it was investigation that turned into premature clean-up, that got de-prioritised.

I think deleting any of these works is the wrong thing to do, if this is actually important, I can schedule finishing it. I'll need to sometime to investigate where I left off. Orphans and dupes have been the traditional pain points, this version of the problem has not up up as a concrete pain point as far as I'm aware.

**Comment by mekarpeles** [comment]
We're going to close this one in favor of #2727

Confirmed that the original work bot is no longer introducing these problems. The rest is cleanup.
